src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkCustomCommandLine.java

```diff
@@ -216,18 +234,22 @@ private static void push(
       features |= arg.joinWith != null ? HAS_JOIN_WITH : 0;
       features |= arg.formatJoined != null ? HAS_FORMAT_JOINED : 0;
       features |= arg.terminateWith != null ? HAS_TERMINATE_WITH : 0;
-      arguments.add(VectorArg.create(features, arg.nestedSetStringificationType));
+      features |= arg.nestedSet == null && arg.list.size() == 1 ? HAS_SINGLE_ARG : 0;
+      arguments.add(
+          VectorArg.create(features, arg.nestedSetStringificationType, starlarkSemantics));
       if (arg.mapEach != null) {
         arguments.add(arg.mapEach);
```

Comment:
I could see interning this as well be beneficial if it is *not* a closure. What do you think?

Reply:
Agree that could be beneficial. Are you able to easily tell if it's a closure? Please make it a separate PR if you proceed with it.

Reply:
It is. I will send a PR after we get the benchmark data for this change.

We could even consider interning `Location` while we are at it. But let's consider that during the review of the follow-up.